apps/backend/src/api/core/helpers/get-balance/index.ts

```diff
@@ -10,19 +10,33 @@ import { ISorobanService, SimulateContractOperation } from 'interfaces/soroban/t
 
 export const getWalletBalance = async ({
   userContractAddress,
+  assetType,
+  assetCode,
   assetRepository,
   sorobanService,
 }: {
   userContractAddress: string
+  assetType?: string
+  assetCode?: string
   assetRepository?: AssetRepositoryType
   sorobanService?: ISorobanService
 }): Promise<number> => {
   const assetRepositoryInstance = assetRepository || AssetRepository.getInstance()
   const sorobanServiceInstance = sorobanService || SorobanService.getInstance()
-  // Get asset contract address from db
-  const asset = await assetRepositoryInstance.getAssetByType('native') // Stellar/XLM native asset
-  const assetContractAddress = asset?.contractAddress ?? STELLAR.TOKEN_CONTRACT.NATIVE
 
+  // Get asset contract address
+  const asset =
+    assetType || assetCode
```

Comment:
```suggestion
    assetType
```

I think it should be this way